contract_forecast/hooks.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api
+from odoo.tools import SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    """
+    Generate contract line forecast periods
+    """
+    _logger.info(
+        "Post init hook for module post_init_hook: "
+        "Generate contract line forecast periods"
+    )
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        contract_lines = env["account.analytic.invoice.line"].search([])
+        contract_lines.with_delay()._generate_forecast_periods()
```

Comment:
@sbejaoui this creates one job that will take ages to run in a database loaded with many lines.

You could improve it in several ways:
- search where is_canceled = False
- search where auto_renew is False and end_date is null or in the future
- create one job per line (or per contract)